app/src/main/java/ch/epfl/sdp/blindwar/domain/game/GameSound.kt

```diff
@@ -1,29 +1,71 @@
 package ch.epfl.sdp.blindwar.domain.game
 
+import android.content.ContentResolver
 import android.content.res.AssetFileDescriptor
 import android.content.res.AssetManager
 import android.media.MediaMetadataRetriever
 import android.media.MediaPlayer
+import androidx.documentfile.provider.DocumentFile
 import ch.epfl.sdp.blindwar.data.sound.LocalSoundDataSource
+import java.io.FileDescriptor
 import java.util.*
 
-class GameSound(assetManager: AssetManager) {
+class GameSound(val assetManager: AssetManager) {
     private val mediaMetadataRetriever = MediaMetadataRetriever()
     private val localSoundDataSource = LocalSoundDataSource(assetManager, mediaMetadataRetriever)
     private val player = MediaPlayer()
+
     // Map each title with its asset file descriptor and its important metadata
-    private lateinit var assetFileDescriptor: Map<String, Pair<AssetFileDescriptor, SongMetaData>>
+    // File descriptor for files in local storage
+    private lateinit var assetFileDescriptorAndMetaDataPerTitle: Map<String, Pair<AssetFileDescriptor, SongMetaData>>
+
+    // File descriptor for files in assets folder
+    private lateinit var fileDescriptorAndMetaDataPerTitle: Map<String, Pair<FileDescriptor, SongMetaData>>
     private lateinit var playlistNames: MutableSet<String>
     private lateinit var currentMetaData: SongMetaData
 
-    fun soundInit(playlist: List<SongMetaData>) {
-        assetFileDescriptor = localSoundDataSource.fetchSoundFileDescriptors(playlist)
+    fun soundInitWithSpotifyMetadata(playlist: List<SongMetaData>) {

```

Comment:
The naming of the function could be confusing as it could refer to either the model song metadata or the API response metadata defined in SpotifyResponse